src/vue/forms/AssetForm.vue

```diff
@@ -261,28 +368,104 @@ export default {
       return this.request instanceof AssetUpdateRequestRecord
     },
   },
-  created () {
-    this.tryPopulateForm(this.request)
+  async created () {
+    if (this.assetForUpdate) {
```

Comment:
You've cleaned up the code well, but haven't solved the problem.

This component still takes too much responsibility. The actions on creating/updating asset request, and actions on creating/updating asset itself are COMPLETELY different actions for the user as well as for the developer.

I still insist on splitting the `AssetForm` to `AssetCreateForm` and `AssetUpdateForm` form and do not try to sit on 2 chairs. This actions have very **SIMILAR** but not the **SAME** flows, so if you'll have to duplicate some flow code between `AssetCreateForm` and `AssetUpdate`, you won't actually duplicate it, you'll make it consistent.

Reply:
`AssetForm` was splitted into two forms: `AssetCreateForm` (for asset creation requests) and `AssetUpdateForm` (for asset update requests). `fromAssetRecord` method is redundant now, so it's been removed.

@max-nes @quinn71 @letehaha